interfaces/Portalicious/src/app/pages/project-registrations/components/change-status-dialog/change-status-dialog.component.ts

```diff
@@ -199,11 +197,18 @@ export class ChangeStatusDialogComponent {
         }
         this.dialogVisible.set(false);
         this.toastService.showToast({
-          summary: $localize`:@@generic-success:Success`,
-          detail: $localize`${data.applicableCount} registration(s) were ${this.statusVerb().toLowerCase()} successfully. The status change can take up to a minute to process.`,
-          severity: 'success',
+          summary: $localize`Changing statuses`,
+          detail: $localize`The status of ${data.applicableCount} registration(s) is being changed to "${this.statusLabel()}" successfully. The status change can take up to a minute to process.`,
+          severity: 'info',
+          showSpinner: true,
         });
+        this.onActionComplete.emit();
         void this.registrationApiService.invalidateCache(this.projectId);
+
+        setTimeout(() => {
+          // invalidate the cache again after a delay to try and make the status change reflected in the UI
+          void this.registrationApiService.invalidateCache(this.projectId);
+        }, 3000);
```

Comment:
Is it by 'design' that we do it twice now? And probably 500 ms or even less is fine.
This action will happen in bulk and we've added the spinner now anyway.

Reply:
> Is it by 'design' that we do it twice now?
Yup

> And probably 500 ms or even less is fine.
Changed it to 500ms